global_planner/src/orientation_filter.cpp

```diff
@@ -35,13 +35,27 @@
  * Author: David V. Lu!!
  *********************************************************************/
 #include <global_planner/orientation_filter.h>
+#include <tf/tf.h>
 #include <tf2/LinearMath/Matrix3x3.h>
 #include <tf2/utils.h>
 #include <tf2_geometry_msgs/tf2_geometry_msgs.h>
 #include <angles/angles.h>
 
 namespace global_planner {
 
+/**
+ * Minimum angle between pose orientations. Assumes both poses are in the same reference frame.
+ * @param a pose a
+ * @param b pose b
+ * @return minimum angle
+ */
+double min_angle(const geometry_msgs::PoseStamped& a, const geometry_msgs::PoseStamped& b)
+{
+    tf::Quaternion qa(a.pose.orientation.x, a.pose.orientation.y, a.pose.orientation.z, a.pose.orientation.w);
+    tf::Quaternion qb(b.pose.orientation.x, b.pose.orientation.y, b.pose.orientation.z, b.pose.orientation.w);
+    return tf::angleShortestPath(qa, qb);
+}
```

Comment:
How about something like this (bonus: you can ignore tf dependency):

```suggestion
double min_angle(const geometry_msgs::PoseStamped& a, const geometry_msgs::PoseStamped& b)
{
    return angles::shortest_angular_distance(tf2::getYaw(a.pose.orientation), tf2::getYaw(b.pose.orientation));
}
```